gRPC/FFmpeg_wrapper_client.cc

```diff
@@ -90,3 +93,111 @@ void CmdPassClient::AsyncCompleteRpc() {
+}
+
+// Function to convert MCM to vector of string pairs
+void mcmToStringPairs(const MCM& mcm, std::vector<std::pair<std::string, std::string>>& result, const std::string& prefix) {
+    result.push_back({prefix + "conn_type", mcm.conn_type});
+    result.push_back({prefix + "transport", mcm.transport});
+    result.push_back({prefix + "transport_pixel_format", mcm.transport_pixel_format});
+    result.push_back({prefix + "ip", mcm.ip});
+    result.push_back({prefix + "port", to_string<int>(mcm.port)});
+    result.push_back({prefix + "urn", mcm.urn});
+}
+
+// Function to convert Payload to vector of string pairs
+void payloadToStringPairs(const Payload& payload, std::vector<std::pair<std::string, std::string>>& result, const std::string& prefix) {
+    result.push_back({prefix + "payload_type", to_string<int>(payload.type)});
+    if (payload.type == video) {
+        videoToStringPairs(payload.video, result, prefix);
+    } else if (payload.type == audio) {
+        audioToStringPairs(payload.audio, result, prefix);
+    }
+}
+
+// Function to convert StreamType to vector of string pairs
+void streamTypeToStringPairs(const StreamType& streamType, std::vector<std::pair<std::string, std::string>>& result, const std::string& prefix) {
+    result.push_back({prefix + "stream_type", to_string<int>(streamType.type)});
+    if (streamType.type == file) {
+        fileToStringPairs(streamType.file, result, prefix);
+    } else if (streamType.type == st2110) {
+        st2110ToStringPairs(streamType.st2110, result, prefix);
+    } else if (streamType.type == mcm) {
+        mcmToStringPairs(streamType.mcm, result, prefix);
+    }
+}
+
+// Function to convert Stream to vector of string pairs
+void streamToStringPairs(const Stream& stream, std::vector<std::pair<std::string, std::string>>& result, const std::string& prefix) {
+    payloadToStringPairs(stream.payload, result, prefix);
+    streamTypeToStringPairs(stream.stream_type, result, prefix);
+}
+
+// Function to convert Config to vector of string pairs
+std::vector<std::pair<std::string, std::string>> commitConfigs(const Config& config) {
+    std::vector<std::pair<std::string, std::string>> result;
+
+    std::string json_str;
+    if(serialize_config_json(config, json_str) != 0) {
+        std::cout << "Error serializing Config" << std::endl;
+    }; 
+    result.push_back({"json", json_str});
+
+    // result.push_back({"function", config.function});
+    // result.push_back({"gpu_hw_acceleration", config.gpu_hw_acceleration});
+    // result.push_back({"logging_level", to_string<int>(config.logging_level)});
+    // for (size_t i = 0; i < config.senders.size(); ++i) {
+    //     streamToStringPairs(config.senders[i], result, "sender_" + to_string<int>(i) + "_");
+    // }
+    // for (size_t i = 0; i < config.receivers.size(); ++i) {
+    //     streamToStringPairs(config.receivers[i], result, "receiver_" + to_string<int>(i) + "_");
+    // }
+
+    return result;
+}
```

Comment:
if we gonna use JSON string so maybe we can remove these commented out lines and remove the other functions as they're obsolete right now.

Reply:
Fixed in,  https://github.com/OpenVisualCloud/Intel-Tiber-Broadcast-Suite/pull/49/commits/a25878f7b55f8de4ffe4b4d7121ad3707906d3c8
In case of json serialization fails, previous method will be used